### DISCUSSION TITLE
Why brew uninstall does not remove data from /opt/homebrew/var

### DISCUSSION THREAD

**Discussion by fatadel** [Homebrew/discussions#3264] [Getting started] [answered]
I have uninstalled the previous version of PostgreSQL (`brew uninstall postgresql@13`), and it is indeed removed now. However, this didn't cleanup related files in the `var` directory, and I still have files in `/opt/homebrew/var/postgresql@13`. Is that intended? Should I manually remove those data and is it safe?

**Comment by gromgit** [comment, marked as answer]
> Is that intended?

Yes, that's the default location of your PostgreSQL database. If `brew uninstall` automatically wiped that, you will be Very Unhappy indeed.

>Should I manually remove those data and is it safe?

You can `rm -fr /opt/homebrew/var/postgresql@13` if you have no more use for your PostgreSQL data.

**Reply by fatadel** [reply]
But I did `brew postgresql-upgrade-database` and `delete_old_cluster.sh` - first one upgrades PosgreSQL data, second one removes it. Logically, all old data should be wiped after this. So, I insist `rm -rf /opt/homebrew/var/postgresql@13` must be a part of this process (apparently a part of `delete_old_cluster.sh`).

**Reply by SMillerDev** [reply]
I'm not sure where `delete_old_cluster.sh` comes from, but it's not from Homebrew

**Reply by gromgit** [reply]
I'm not a maintainer, but I would be Very Angry Indeed if `brew uninstall <formula>` automatically wiped all data directories used by `<formula>`. Remember:
1. People make misteaks all the time.
2. If a `brew reinstall <formula>` doesn't fix things for some reason, people naturally try `brew uninstall` followed by a `brew install`. These people definitely _don't_ expect their databases to suddenly disappear, because...
3. No package manager in the known universe nukes user data on a package uninstall, _even if the install created the data directories for that package_.

If you can show just _one_ example of a `(yum|apt|deb|port|etc.) uninstall postgresql` that automatically deletes a PostgreSQL data directory, the Homebrew maintainers might reconsider...or not.
